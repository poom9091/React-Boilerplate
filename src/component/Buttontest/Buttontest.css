.button-style{
    border: 0.2rem solid black;
    border-radius: 1rem;
    height: max-content;
    margin: auto;
    padding: 0.8rem;
    cursor: pointer;
}