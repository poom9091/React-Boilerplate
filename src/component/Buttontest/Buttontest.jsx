import React from 'react'
import './Buttontest.css'
function Buttontest({label}) {
    return (
        <div data-testid="button" className="button-style">
            {label}
        </div>
    )
}

export default Buttontest
