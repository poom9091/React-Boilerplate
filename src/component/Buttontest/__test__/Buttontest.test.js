import React from 'react';
import ReactDOM from 'react-dom';
import Buttontest from '../Buttontest';

import {render,cleanup} from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer"

afterEach(cleanup);

it("render without crashing",()=>{
    const button = document.createElement("button")
    ReactDOM.render(<Buttontest/>, button)
    // ReactDOM.render
})

it("render button correctly",() => {
    const {getByTestId} = render(<Buttontest label="click me plase" />)
    expect(getByTestId('button')).toHaveTextContent("click me plase")
})

it("render button correctly",() => {
    const {getByTestId} = render(<Buttontest label="click me plase" />)
    expect(getByTestId('button')).toHaveTextContent("click me plase")
})

it("matches snapshot",()=>{
    const tree = renderer.create(<Buttontest label="save"/>).toJSON();
    expect (tree).toMatchSnapshot();
})


it("matches snapshot",()=>{
    const tree = renderer.create(<Buttontest label="click me please"/>).toJSON();
    expect (tree).toMatchSnapshot();
})