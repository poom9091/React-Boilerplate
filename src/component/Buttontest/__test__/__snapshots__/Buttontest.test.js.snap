// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="px-4 py-2 bg-blue-600 rounded-lg shadow-lg text-white "
  data-testid="button"
>
  save
</button>
`;

exports[`matches snapshot 2`] = `
<button
  className="px-4 py-2 bg-blue-600 rounded-lg shadow-lg text-white "
  data-testid="button"
>
  click me please
</button>
`;
